crates/bevy_winit/src/system.rs

```diff
@@ -0,0 +1,245 @@
+use bevy_ecs::{
+    entity::Entity,
+    event::EventReader,
+    prelude::{Added, Changed, With},
+    system::{Commands, NonSendMut, Query, Res},
+};
+use bevy_utils::tracing::{error, info};
+use bevy_window::{
+    Cursor, CursorPosition, PrimaryWindow, Window, WindowClosed, WindowComponents, WindowHandle,
+    WindowMode, WindowPosition, WindowResizeConstraints, WindowResolution, WindowState,
+    WindowTitle,
+};
+use raw_window_handle::HasRawWindowHandle;
+use winit::{
+    dpi::{LogicalSize, PhysicalPosition},
+    event_loop::EventLoopWindowTarget,
+};
+
+use crate::{converters, get_best_videomode, get_fitting_videomode, WinitWindows};
+
+/// System responsible for creating new windows whenever a `Window` component is added
+/// to an entity.
+///
+/// This will default any necessary components if they are not already added.
+pub fn create_window_system(
+    mut commands: Commands,
+    event_loop: &EventLoopWindowTarget<()>,
+    created_windows: Query<(Entity, WindowComponents), Added<Window>>,
+    mut winit_windows: NonSendMut<WinitWindows>,
+) {
+    for (window_entity, components) in &created_windows {
+        if let Some(_) = winit_windows.get_window(window_entity) {
+            // Just a safe guard
+            continue;
+        }
+
+        info!("Creating a new window: {:?}", window_entity);
+
+        let winit_window = winit_windows.create_window(&event_loop, window_entity, &components);
+
+        commands
+            .entity(window_entity)
+            .insert(WindowHandle::new(winit_window.raw_window_handle()));
```

Comment:
After winit has created the window, the `WindowCreated` event should be emitted. I don't see anything that is currently emitting the `WindowCreated` event

Reply:
Always forgot to do this, should be done now, ty for the comment!